.github/workflows/unit_tests.yaml

```diff
@@ -0,0 +1,54 @@
+name: Run unit tests
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: hammer-cli-foreman-puppet
+    strategy:
+      fail-fast: false
+      matrix:
+        #ruby: [2.3, 2.4, 2.5]
+        ruby: [2.5]
+
+    steps:
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Get hammer-cli
+        uses: actions/checkout@v2
+        with:
+          repository: theforeman/hammer-cli
+          ref: master
+          path: hammer-cli
+      - name: Get hammer-cli-foreman
+        uses: actions/checkout@v2
+        with:
+          repository: amirfefer/hammer-cli-foreman
+          ref: puppet-extraction
+          path: hammer-cli-foreman
+      - name: Get hammer-cli-foreman-puppet
+        uses: actions/checkout@v2
+        with:
+          path: hammer-cli-foreman-puppet
+      - name: Configure local gem dependencies
+        run: |
+          ls -al ..
+          ls -al
+          pwd
+          echo "gemspec path: '../hammer-cli-foreman', name: 'hammer_cli_foreman'" > Gemfile.local.rb
+      - name: Install dependencies
+        run: |
+          pwd
+          bundle install
+          #          gem update bundler
```

Comment:
If you call `setup-ruby` at this point (with `bundler-cache: true`) you actually get bundle install with cache.